mayday/file_test.go

```diff
@@ -1,35 +1,50 @@
 package mayday_test
 
 import (
+	"bytes"
+	"io"
 	"io/ioutil"
+	"os"
 	"testing"
 
 	"github.com/coreos/mayday/mayday"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestNonexistentFile(t *testing.T) {
-	file := mayday.File{Path: "/etc/nonexistent"}
-	err := file.Collect(workspace)
-	assert.Equal(t, err.Error(), `error opening source file: open /etc/nonexistent: no such file or directory`)
+type tempReadCloser struct {
+	io.Reader
+}
+
+func (tempReadCloser) Close() error {
+	return nil
 }
 
 func TestFile(t *testing.T) {
-	file := mayday.File{Path: "/proc/meminfo"}
+
+	file := mayday.File{
+		Source: tempReadCloser{bytes.NewBufferString("contents1")},
```

Comment:
This only uses the temp dir as the string for the path. `tempReadCloser` is perhaps a bad name, but it literally just uses the `Buffer` returned by `NewBufferString` to store the "file" data.